app/scripts/BaseClasses/BaseController.coffee

```diff
@@ -7,6 +7,14 @@
 
 module.exports = class BaseCtrl
 
+# all dataSets
+  @fileSets = ['AlzheimerNeuroimagingData_projector_config.json', 'Antarctic_Ice_Thickness_projector_config.json', 'Baseball_Players_projector_config.json',
+              'BiomedBigMetadata_projector_config.json', 'California_Ozone_Pollution_projector_config.json', 'California_Ozone_projector_config.json',
+              'Countries_Rankings_projector_config.json', 'Fortune500_projector_config.json', 'HeartAttacks_projector_config.json',
+              'SOCR_CountryRanking_projector_config.json', 'SOCR_UKBB_projector_config.json', 'SchizophreniaNeuroimaging_projector_config.json',
+              'Turkiye_Student_Evaluation_Data_Set_projector_config.json', 'US_Ozone_Pollution_projector_config.json',
+              'iris_projector_config.json', 'knee_pain_data_projector_config.json']
+
```

Comment:
This class is a Base class representing abstract Controller and should only have methods and properties that al Controllers have to inherit. These datasets should be moved to the class of the corresponding Controller.

Reply:
You can consider creating a separate Service that provides a list of the available data files and include it as a dependency for the corresponding Controller (just not the Base one).

Reply:
Hi, I just moved this inside DimensionReductionSidebarCtrl.ctrl.coffee. Please see the updated code.